app/services/feeds/optimized.rb

```diff
@@ -0,0 +1,279 @@
+module Feeds
+  class Optimized
+    RANDOM_OFFSET_VALUES = [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13].freeze
+
+    def initialize(user: nil, number_of_articles: 35, page: 1, tag: nil)
+      @user = user
+      @number_of_articles = number_of_articles
+      @page = page
+      @tag = tag
+      @randomness = 3 # default number for randomly adjusting feed
+      @tag_weight = 1 # default weight tags play in rankings
+      @comment_weight = 0 # default weight comments play in rankings
+      @experience_level_weight = 1 # default weight for user experience level
+    end
+
+    def self.find_featured_story(stories)
+      featured_story =  if stories.is_a?(ActiveRecord::Relation)
+                          stories.where.not(main_image: nil).first
+                        else
+                          stories.detect { |story| !story.main_image.nil? }
```

Comment:
IMHO this reads better than the negated version:

```suggestion
                          stories.detect { |story| story.main_image.present? }
```